core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadAccessParser.java

```diff
@@ -60,10 +60,32 @@ public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay readerWay, boolean fer
         SpatialRuleSet spatialRuleSet = readerWay.getTag("spatial_rule_set", null);
         if (spatialRuleSet != null && spatialRuleSet != SpatialRuleSet.EMPTY) {
             RoadClass roadClass = RoadClass.find(readerWay.getTag("highway", ""));
-            accessValue = spatialRuleSet.getAccess(roadClass, TransportationMode.MOTOR_VEHICLE, YES);
+            accessValue = spatialRuleSet.getAccess(roadClass, TransportationMode.CAR, YES);
         }
-            
+
         roadAccessEnc.setEnum(false, edgeFlags, accessValue);
         return edgeFlags;
     }
+
+    public static List<String> toOSMRestrictions(TransportationMode mode) {
+        switch (mode) {
+            case FOOT:
+                return Arrays.asList("foot", "access");
+            case VEHICLE:
+                return Arrays.asList("vehicle", "access");
+            case BIKE:
+                return Arrays.asList("bicycle", "vehicle", "access");
+            case MOTOR_VEHICLE:
+                return Arrays.asList("motor_vehicle", "vehicle", "access");
+            case CAR:
+                return Arrays.asList("motorcar", "motor_vehicle", "vehicle", "access");
+            case MOTORCYCLE:
+                return Arrays.asList("motorcycle", "motor_vehicle", "vehicle", "access");
+            case HGV:
+                return Arrays.asList("hgv", "motor_vehicle", "vehicle", "access");
+            default:
+                return mode.isMotorVehicle() ?
```

Comment:
Shouldn't we add a case for each mode and throw an exception in default? We could add a unittest to ensure that all TransportationModes return a value for `toOSMRestrictions()`

Reply:
Good question. Having this default might make it easier for forks but forcing a complete list of TransportationModes also sounds good.

Reply:
I'd argue that if a fork wants to extend TransportMode they need think about what restrictions are appropriate.

Reply:
The problem is, that if you can't change this when using GraphHopper as a lib, you would actually have to fork it.

Reply:
I do not see a big problem that this is not customizable (it wasn't before and there are currently no use cases known that need this).

Reply:
TransportationMode is an enum so it can't be extended by library users without creating a fork?